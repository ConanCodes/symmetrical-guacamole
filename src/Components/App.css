.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

html {
  font-family: sans-serif;
}

body {
  background: white;
}

h2 {
  font-size: 1.25rem;
  margin: 1rem 0.5rem;
}

progress {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 80%;
  height: 2rem;
  color: #ff0062;
  border: 1px solid #a927f5;
}

progress[value]::-webkit-progress-bar,
progress[value] {
  background-color: #eee;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25) inset;
}

progress[value]::-webkit-progress-value {
  background-color: #a927f5;
  transition: all 0.2s ease-out;
}

progress[value]::-moz-progress-bar {
  background-color: #a927f5;
  transition: all 0.2s ease-out;
}

button {
  font: inherit;
  background: #ff0062;
  border: 1px solid #ff0062;
  color: white;
  padding: 1rem 2rem;
  border-radius: 6px;
  cursor: pointer;
  margin: 0.5rem;
}

button:focus {
  outline: none;
}

button:hover,
button:active {
  background: #a927f5;
  border-color: #a927f5;
}

#health-levels,
#controls {
  margin: 2rem auto;
  width: 30rem;
  text-align: center;
  border-radius: 10px;
  padding: 1rem;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.25);
  background: white;
}

#controls {
  display: flex;
  flex-direction: column;
}

#bonus-life {
  font-weight: bold;
  color: white;
  background: #ff0062;
  border: 1px solid #ff0062;
  padding: 0.15rem 0.5rem;
  border-radius: 10px;
  text-align: center;
  margin: 0 0.5rem;
}
