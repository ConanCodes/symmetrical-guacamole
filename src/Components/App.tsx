import './App.css';
import { PureComponent } from 'react';
import Navbar from "./Navbar";

// I dont know why but this import is causing issues, Ill look into it later i guess
// import logo from '../assets/logo.svg';

/** Advice/Todo
 * Rule of thumb everything should be placed withing your App div
 * 
 * <div className="app">
 * <NavBar/>
 * ...etc
 * </div>
 * 
 * Doing this will allow you to controll the whole app when we start learning state management
 * Putting it in the div will allow you to remove the empty tags <> which will be less for the the dom to render, so better performance in the long run
 */


/**
 * If our NavBar is above our app (not in the div) then we dont have to declare it in each module
 */

export default class App extends PureComponent {

  render() {
    return (
      <><Navbar />
        <div className="App">
          <header className="App-header">
            {/* <img src={logo} className="App-logo" alt="logo" /> */}
          </header>
        </div>
      </>
    )
  }
}