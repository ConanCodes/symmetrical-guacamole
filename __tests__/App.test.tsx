import { createRoot } from 'react-dom/client';

import { act } from "react-dom/test-utils";
import App from '../src/components/App';
import { MemoryRouter as Router } from 'react-router-dom'
import BasicTable from '../src/components/BasicTable';
import React from 'react';
const pretty = require('pretty');

let container: HTMLElement | null;
let root: any;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  root = createRoot(container);
  container = document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  root.unmount();
  container?.remove();
  container = null;
});

test("App", () => {
  act(() => { root.render(<Router><App /></Router>); });
  expect(pretty(container?.innerHTML)).toMatchSnapshot()
});

test("basicTable", () => {
  act(() => { root.render(<Router><BasicTable /></Router>); });
  expect(pretty(container?.innerHTML)).toMatchSnapshot()
});