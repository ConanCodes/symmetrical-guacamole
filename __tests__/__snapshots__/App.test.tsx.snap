// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
"<nav class=\\"nav\\"><a href=\\"/\\" class=\\"/\\">Home</a><a href=\\"/ButtonPage\\">ButtonPage</a><a href=\\"/BasicTable\\">BasicTable</a><a href=\\"/guacc\\">Guacc</a><a href=\\"/howtobowl\\">HowToBowl</a></nav>
<div class=\\"App\\">
  <header class=\\"App-header\\"></header>
</div>"
`;

exports[`basicTable 1`] = `
"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root\\">
  <table class=\\"MuiTable-root css-quj9j6-MuiTable-root\\" aria-label=\\"simple table\\">
    <thead class=\\"MuiTableHead-root css-15wwp11-MuiTableHead-root\\">
      <tr class=\\"MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root\\">
        <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Dessert (100g serving)</th>
        <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root\\" scope=\\"col\\">Calories</th>
        <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root\\" scope=\\"col\\">Fat&nbsp;(g)</th>
        <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root\\" scope=\\"col\\">Carbs&nbsp;(g)</th>
        <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root\\" scope=\\"col\\">Protein&nbsp;(g)</th>
      </tr>
    </thead>
    <tbody class=\\"MuiTableBody-root css-apqrd9-MuiTableBody-root\\">
      <tr class=\\"MuiTableRow-root css-34nofg-MuiTableRow-root\\">
        <th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\" scope=\\"row\\">Frozen yoghurt</th>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">159</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">6</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">24</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">4</td>
      </tr>
      <tr class=\\"MuiTableRow-root css-34nofg-MuiTableRow-root\\">
        <th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\" scope=\\"row\\">Ice cream sandwich</th>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">237</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">9</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">37</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">4.3</td>
      </tr>
      <tr class=\\"MuiTableRow-root css-34nofg-MuiTableRow-root\\">
        <th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\" scope=\\"row\\">Eclair</th>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">262</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">16</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">24</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">6</td>
      </tr>
      <tr class=\\"MuiTableRow-root css-34nofg-MuiTableRow-root\\">
        <th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\" scope=\\"row\\">Cupcake</th>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">305</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">3.7</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">67</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">4.3</td>
      </tr>
      <tr class=\\"MuiTableRow-root css-34nofg-MuiTableRow-root\\">
        <th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\" scope=\\"row\\">Gingerbread</th>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">356</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">16</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">49</td>
        <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root\\">3.9</td>
      </tr>
    </tbody>
  </table>
</div><a href=\\"/\\">Home</a>"
`;
